plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -370,6 +373,37 @@ public Transaction newCreateTableTransaction(
                 Optional.of(session.getUser()));
     }
 
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String metadataLocation)
+            throws TrinoException
+    {
+        if (tableExists(schemaTableName)) {
+            throw new TrinoException(TABLE_ALREADY_EXISTS, format("Table '%s' already exists", schemaTableName));
+        }
```

Comment:
- Do we have a test for this? (against Glue catalog)
- If we take out these lines, would it still pass?
  - if we do one call to Glue with proper err handling, we won't have any concurrency issues

Reply:
Yes, We have test against Glue catalog as well.